### PR TITLE
[SYCLomatic] Fix thrust API migration bug of missing template arguments

### PR DESCRIPTION
**Pull request by delecui**

### CODE REVIEW

**Review by AndyCHHuang — APPROVED** [review]
LGTM

**Review by tomflinda — APPROVED** [review]
LGTM

File: clang/lib/DPCT/ASTTraversal.cpp [hunk 1]
```diff
@@ -2529,7 +2540,8 @@ void TypeInDeclRule::registerMatcher(MatchFinder &MF) {
                   "cooperative_groups::__v1::thread_block",
                   "libraryPropertyType_t", "libraryPropertyType",
                   "cudaDataType_t", "cudaDataType", "cublasComputeType_t",
-                  "cublasAtomicsMode_t", "CUmem_advise_enum", "CUmem_advise"),
+                  "cublasAtomicsMode_t", "CUmem_advise_enum", "CUmem_advise",
+                  "thrust::tuple_element"),
```
Comment (ziranzha) [inline clang/lib/DPCT/ASTTraversal.cpp]: Feature support should not combined with bug fix.

Comment (delecui) [reply clang/lib/DPCT/ASTTraversal.cpp]: OK, got it.
The current PR is for bug fix.
I will move the feature "support migration of tuple_element" to another pull request

**Review by ziranzha — APPROVED** [review]

**Pull request merged**